CMake/external_fastdds.cmake

```diff
@@ -0,0 +1,84 @@
+                   -DCMAKE_C_FLAGS_RELWITHDEBINFO=${CMAKE_C_FLAGS_RELWITHDEBINFO}
+                   -DCMAKE_CXX_STANDARD_LIBRARIES=${CMAKE_CXX_STANDARD_LIBRARIES}
+                   -DBUILD_SHARED_LIBS:BOOL=OFF
+                   -DFOONATHAN_MEMORY_BUILD_EXAMPLES:BOOL=OFF
+                   -DFOONATHAN_MEMORY_BUILD_TESTS:BOOL=OFF
+                   -DFOONATHAN_MEMORY_BUILD_TOOLS:BOOL=OFF
+                   -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/fastdds/fastdds_install
+        UPDATE_COMMAND ""
+        PATCH_COMMAND ""
+        TEST_COMMAND ""
+        )
+    
+    
+    # Let FastDDS clone and build it's dependencies (except "Foonathan_memory" which fastdds does not supply as a third party.
+    set(FASTDDS_FLAGS   -DBUILD_SHARED_LIBS=OFF 
+                        -DTHIRDPARTY_Asio=FORCE 
+                        -DTHIRDPARTY_TinyXML2=FORCE 
+                        -DTHIRDPARTY_fastcdr=FORCE)
+                            
+    # We construct the git tag is the purpose of having a single place that indicate the FastDDS version we consume.
+    # FastDDS library name is different in Windows/Linux (Windows library name is versioned and Linux is not!)
+    #   Windows: libfastrtps-<major-version>.<minor-version>.lib
+    #   Linux:   libfastrtps.a 
+    # Current used version is "2.5.0"
+    # Commit Hash: ecb9711cf2b9bcc608de7d45fc36d3a653d3bf05
+    # Git tag "v2.5.0"
+    set(FASTDDS_VERSION_MAJOR "2")
+    set(FASTDDS_VERSION_MINOR "5")
+    set(FASTDDS_VERSION_BUILD "0")
+    
+    ExternalProject_Add(
+        fastdds
+        PREFIX fastdds
+        GIT_REPOSITORY "https://github.com/eProsima/Fast-DDS.git"
+        GIT_TAG "v${FASTDDS_VERSION_MAJOR}.${FASTDDS_VERSION_MINOR}.${FASTDDS_VERSION_BUILD}" 
+        SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/third-party/fastdds
+        CMAKE_ARGS  -DCMAKE_CXX_FLAGS=${CMAKE_CXX_FLAGS}
+                    -DCMAKE_C_FLAGS=${CMAKE_C_FLAGS}
+                    -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+                    -DCMAKE_C_FLAGS_DEBUG=${CMAKE_C_FLAGS_DEBUG}
+                    -DCMAKE_C_FLAGS_MINSIZEREL=${CMAKE_C_FLAGS_MINSIZEREL}
+                    -DCMAKE_C_FLAGS_RELEASE=${CMAKE_C_FLAGS_RELEASE}
+                    -DCMAKE_C_FLAGS_RELWITHDEBINFO=${CMAKE_C_FLAGS_RELWITHDEBINFO}
+                    -DCMAKE_CXX_STANDARD_LIBRARIES=${CMAKE_CXX_STANDARD_LIBRARIES}
+                    -DCMAKE_INSTALL_PREFIX=${CMAKE_CURRENT_BINARY_DIR}/fastdds/fastdds_install
+                    -DCMAKE_INSTALL_LIBDIR=lib
+                    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE}
+                    ${FASTDDS_FLAGS}
+        UPDATE_COMMAND ""
+        PATCH_COMMAND ""
+        TEST_COMMAND ""
+        DEPENDS "foonathan_memory"
+    )
+    
+    set(FASTDDS_DEBUG_TARGET_NAME "libfastrtpsd")
+    set(FASTDDS_RELEASE_TARGET_NAME "libfastrtps")
+
+    if(WIN32)
+        set(FASTDDS_DEBUG_TARGET_NAME "${FASTDDS_DEBUG_TARGET_NAME}-${FASTDDS_VERSION_MAJOR}.${FASTDDS_VERSION_MINOR}")
```

Comment:
I assume you've tried building in all release types? Debug, Release, ReleaseWithDebug?

Reply:
Good think you asked,
I tested:
Win  Debug, Release, ReleaseWithDebug
Linux: Release

all worked,
Not tried Linux debug and the name is different :(
I am on it.

Reply:
https://github.com/eProsima/Fast-DDS/blob/94f9ed869871e21161a722e6709532e756f2b5a2/src/cpp/CMakeLists.txt#L486
They rename the lib on windows... will fix it